Задание готово, все работает

ProjectKredos opened the pull request:
Задание готово

ProjectKredos committed:
Задание готово, все работает

SeveNChaK commented on src/main/java/goods.java:
```diff
@@ -0,0 +1,36 @@
+import java.util.Scanner;
+
+public class goods {
```
⏫ Название классов принято писать с большой буквы

SeveNChaK commented on src/main/java/Main.java:
```diff
@@ -1,8 +1,22 @@
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("На скольких человек необходимо разделить счёт?");
+        while (true) {
+            countUser = scanner.nextInt();
```
⚠️ Не хватает проверки на не корректный ввод, например, если введут буквы.

SeveNChaK commented on src/main/java/goods.java:
```diff
@@ -0,0 +1,36 @@
+            System.out.println("Введите цену товара");
+            goodsPrice = scanner.nextDouble();
+            System.out.println("Товар " + goodsName + " за " + goodsPrice + " " + rubToText(goodsPrice) + " успешно добавлен. Добавьте еще или завершите." );
+            goodsList = goodsList + "\n" + goodsName + " " + goodsPrice;
```
⏫ Так как `String` - неизменяемый класс, при каждом сложении строк создается новый экземпляр строки. А так как ты не ведешь список товаров, а сразу плюсуешь все в строку, то у тебя будет создавать очень много объектов строковых. Для таких ситуаций лучше использовать `StringBuilder`.
[Про стринг билдер](https://javarush.com/groups/posts/2351-znakomstvo-so-string-stringbuffer-i-stringbuilder-v-java#Класс-StringBuilder)

SeveNChaK commented on src/main/java/goods.java:
```diff
@@ -0,0 +1,36 @@
+            goodsName = scanner.next();
+            if (goodsName.equalsIgnoreCase("Завершить") ) {break;}
+            System.out.println("Введите цену товара");
+            goodsPrice = scanner.nextDouble();
```
⚠️ Так как нет проверки на то что вводится, при вводе букв приложение упадет

SeveNChaK commented on src/main/java/goods.java:
```diff
@@ -0,0 +1,36 @@
+
+    public static String rubToText(double sumRub){
+        if (sumRub<2) { return("рубль");
+        } else if (sumRub>5) { return("рублей");
```
⚠️ Для чисел больше 20, которые оканчиваются на 1-4 здесь используется не правильное склонение - "рублей"

SeveNChaK reviewed (commented):
В некоторых файлах отступы не одинаковые - рекомендую использовать встроенные форматтер. Если ты программируешь в IDE от JetBrains, то можешь пройтись по всем файлам и нажать комбинацию (**Windows**: `Shift + Ctrl + Alt + L`, **MacOS**: `Shift + Option + Command + L`), затем выбрать, какую часть кода хочешь изменить, и выполнить команду. Среда разработки автоматически сделает нужные отступы и переносы (как она это делает можно изменить в настройках, но дефолтные значения вполне нормальные).

ProjectKredos committed:
Исправлены:
1) Корректность склонения "Рубли";
2) Использован String Bilder;
3) Добавлены проверки:
- Ввод текста вместо цифр
- Ввод отрицательных цен
4) Выровнен код через Ctrl + Alt + Shift + L
5) Исправлено написание классов "С большой буквы"

ProjectKredos committed:
Исправлены:
1) Корректность склонения "Рубли";
2) Использован String Bilder;
3) Добавлены проверки:
- Ввод текста вместо цифр
- Ввод отрицательных цен
4) Выровнен код через Ctrl + Alt + Shift + L
5) Исправлено написание классов "С большой буквы"

ProjectKredos committed:
Все одним комитом.

Исправлены:
1) Корректность склонения "Рубли";
2) Использован String Bilder;
3) Добавлены проверки:
- Ввод текста вместо цифр
- Ввод отрицательных цен
4) Выровнен код через Ctrl + Alt + Shift + L
5) Исправлено написание классов "С большой буквы"

ProjectKredos committed:
Все одним комитом.

Исправлены:
1) Корректность склонения "Рубли";
2) Использован String Bilder;
3) Добавлены проверки:
- Ввод текста вместо цифр
- Ввод отрицательных цен
4) Выровнен код через Ctrl + Alt + Shift + L
5) Исправлено написание классов "С большой буквы"

ProjectKredos committed:
Все одним комитом.

Исправлены:
1) Корректность склонения "Рубли";
2) Использован String Bilder;
3) Добавлены проверки:
- Ввод текста вместо цифр
- Ввод отрицательных цен
4) Выровнен код через Ctrl + Alt + Shift + L
5) Исправлено написание классов "С большой буквы"

SeveNChaK commented on src/main/java/Goods.java:
```diff
@@ -0,0 +1,55 @@
+                if (priceScanner.hasNextDouble()) {
+                    goodsPrice = priceScanner.nextDouble();
+                    if (goodsPrice < 0) {
+                        goodsPrice = Math.abs(goodsPrice);
```
Мне кажется, что лучше было бы выводить ошибку и просить повторить ввод. Но это уже так, на поразмышлять)
